empress_gui.py

```diff
@@ -4,362 +4,819 @@
+        self.update_compute_reconciliations_btn()
         return True # return True means allowing the change to happen
 
     def validate_trans_input(self, input_after_change: str):
+        """Transfer cost is only allowed to be a float that is >= 0."""
         try:
             val = float(input_after_change)
             if val >= 0:
                 self.trans_cost = val
+                self.trans_entry_box.set_border_color("green")
             else:
                 self.trans_cost = None   
-                self.trans_error.config(text="should be non-negative", fg="red")    
+                self.trans_entry_box.set_border_color("red")     
         except ValueError:
             self.trans_cost = None
-            self.trans_error.config(text="should be a number", fg="red")
+            self.trans_entry_box.set_border_color("red")
         
-        if self.trans_cost is not None:
-            self.trans_error.config(text="valid", fg="green")
-        self.update_recon_btn()
+        self.update_compute_reconciliations_btn()
         return True # return True means allowing the change to happen
     
     def validate_loss_input(self, input_after_change: str):
+        """Loss cost is only allowed to be a float that is >= 0."""
         try:
             val = float(input_after_change)
             if val >= 0:
                 self.loss_cost = val
+                self.loss_entry_box.set_border_color("green")
             else:
                 self.loss_cost = None   
-                self.loss_error.config(text="should be non-negative", fg="red")    
+                self.loss_entry_box.set_border_color("red") 
         except ValueError:
             self.loss_cost = None
-            self.loss_error.config(text="should be a number", fg="red")
-        
-        if self.loss_cost is not None:
-            self.loss_error.config(text="valid", fg="green")
-        self.update_recon_btn()
+            self.loss_entry_box.set_border_color("red") 
+
+        self.update_compute_reconciliations_btn()
         return True # return True means allowing the change to happen
 
-    def plot_cost_regions(self):
-        """Plots the cost regions using matplotlib and embeds the graph in a tkinter window."""    
+    def update_compute_reconciliations_btn(self):
+        """When the dtl costs inputs are all valid, enable the next button and close unnecessary windows."""
+        if self.dup_cost is not None and self.trans_cost is not None and self.loss_cost is not None:
+            # Enable the next button
+            self.compute_reconciliations_btn.configure(state=tk.NORMAL)
+            self.view_solution_space_dropdown.configure(state=tk.DISABLED)
+            self.view_reconciliations_dropdown.configure(state=tk.DISABLED)
+            self.view_pvalue_histogram_btn.configure(state=tk.DISABLED)
+            
+            if self.entire_space_window is not None and self.entire_space_window.winfo_exists() == 1:
```

Comment:
Write this as `self.entire_space_window is not None and self.entire_space_window.winfo_exists():`. Remove the `== 1`.

Reply:
Done